overlord/snapstate/backend.go

```diff
@@ -60,3 +63,22 @@ func (s *defaultBackend) Rollback(name, ver string, meter progress.Meter) (strin
 func (s *defaultBackend) Activate(name string, active bool, meter progress.Meter) error {
 	return snappy.SetActive(name, active, meter)
 }
+
+func (s *defaultBackend) Download(name, channel string, meter progress.Meter) (string, string, error) {
+	mStore := snappy.NewUbuntuStoreSnapRepository()
+	snap, err := mStore.Snap(name, channel)
+	if err != nil {
+		return "", "", err
+	}
+
+	downloadedSnapFile, err := mStore.Download(snap, meter)
+	if err != nil {
+		return "", "", err
+	}
+
+	if err := snap.SaveStoreManifest(); err != nil {
```

Comment:
maybe a TODO about doing this possibly later? it seems you told me it was done later in current code